lib/ex_unit/lib/ex_unit/runner.ex

```diff
@@ -1,14 +1,30 @@
 defmodule ExUnit.Runner do
   @moduledoc false
 
+  defmodule MaxFailureCounter do
```

Comment:
Should this be `FailuresCounter`? This module counts the number of failures, but it has no knowledge of "maximum number of failures allowed".

I would also rename all `fail_counter` variables to `failures_counter` for consistency.